### ISSUE TITLE
Recreating an account breaks things

### ISSUE THREAD

**Issue by jrieko** [Codewars/codewars.com#467]
Steps to reproduce:
1. Create an account, linked to github
2. Delete the account via the button on account settings
3. Sign up for another account with same github oauth

* The account will have a random username, email address, default avatar
* Will not be linked to a github account
* Clicking the link to github button will just redirect to the profile page. Won't be linked to github. No error shown
* Changing the account email to that used in step 1, then `codewars.com/users/<username>` will display "This user has deleted their profile" regardless of changing username

Basically, the text above the delete account button, "After deleting your account you are free to sign up again using the same email address," isn't very true.

**Comment by jrieko** [comment]
Furthermore, trying to recreate an account with the same email, without github oauth, flat out fails. Email and username "is already taken." So the aforementioned delete account text is completely incorrect.

**Comment by jrieko** [comment]
Then if I try to log in with github oauth, it's successful and takes me to the account that was supposedly deleted. :disappointed:

**Comment by zjkettell** [comment]
I am currently having this same issue, deleted my profile that was connected to my github and went to remake it and it just signed me back into the deleted account, except that account had no profile and default name/email.

**Comment by mmarla** [comment]
Hi, I have the same issue. I deleted an account linked to my Github account and now a) when I try to link github to my new account there's a message that my "github is already linked to another account"; b) the same happens if I delete Codewars from the Github authorized applications list and then attempt to link the accounts; c) I can still login to Codewars using my githib account, which takes me to the deleted account.

**Comment by lorenalexm** [comment]
Popping my head in to say I am experiencing the same issues as all those above me. All the same steps taken, unable to find a way to unassociated and link GitHub account with new Codewars profile.

**Comment by ggpsgeorge** [comment]
Me too, same problems listed above with the same steps

**Comment by jakbyte** [comment]
I created an account linked with GitHub, deleted it, and tried to create one with the same username and email, but it said both were already taken.
